### PR TITLE
Linting and Tailwind setup

### PR DESCRIPTION
**Pull request by CharaD7**
- Added extra linting rules
- Added support for auto-sorting imports in ascending order
- Added support for aliasing imports, for example `import CardInfo from '../../components/Card/CardInfo'` can now be `import CardInfo from '@components/Card/CardInfo'`
- Added support for prettier fixing lints on staging.
- Added tailwind config setup

- [x] Resolved support for aliasing imports, for example `import CardInfo from '../../components/Card/CardInfo'` to `import CardInfo from '~components/Card/CardInfo'`

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+          },
+          {
+            pattern: '@components/*',
+            group: 'internal',
+          },
+          {
+            pattern: '@pages/*',
+            group: 'internal',
+          },
+        ],
```
Comment (b-tiwari) [inline .eslintrc.js:82]: ❤️  loved this, I didn't knew about this patterns can be set as rules

File: .lintstagedrc.js [hunk 2]
```diff
@@ -9,6 +9,10 @@ const path = require('path')
 const buildEslintCommand = (filenames) =>
   `eslint --quiet --fix ${filenames.join(' ')}`
 
+const buildPrettierCommand = (filenames) =>
+  `prettier --loglevel silent --write --ignore-unknown ${filenames.join(' ')}}`
+
```
Comment (b-tiwari) [inline .lintstagedrc.js:15]: 👍

File: jsconfig.json [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+		"moduleResolution": "node",
+		"compilerOptions": {
+				"jsx": "preserve",
+				"baseUrl": ".",
+				"paths": {
+						"@/*": ["*"]
+				}
+		},
+		"exclude": ["node_modules", ".next"]
+}
```
Comment (b-tiwari) [inline jsconfig.json:13]: do we need jsconfig? is this the file that allows to use import path aliases with @?

**Review by b-tiwari — APPROVED** [review]
this is cool, I am glad to have tailwind css for the app.
also, thanks for the new useful linting/formatting rules, these look cool and useful.

File: .eslintrc.js (hunk 1 above)
Comment (CharaD7) [reply .eslintrc.js:82]: I am glad you like it.

File: jsconfig.json (hunk 3 above)
Comment (CharaD7) [reply jsconfig.json:13]: Yes it is, and I have just resolved the @ alias error messages you had. Kindly check fresh pull for confirmation.

Comment (b-tiwari) [reply jsconfig.json:13]: yes, its working now @CharaD7 , I think this PR can be merged now

Comment (b-tiwari) [reply jsconfig.json:13]: actually hold on, I see the auto formatting of code as per our prettier config on save file is not working in this brach.

Comment (b-tiwari) [reply jsconfig.json:13]: looks like 'plugin:jest/recommended' is overriding our prettier rules, removing it from eslintrc's extensions brings back the prettier formatting. 
Do we need this? I think we can remove 'plugin:jest/recommended'

**Comment by b-tiwari** [comment]
@CharaD7 , this PR has been pending to be merged for long, I was expecting it to get reviewed by one more peer, but because this is holding up other work, I am going to merge this now

**Pull request merged**